tooltester/src/main/java/io/dockstore/tooltester/client/cli/Client.java

```diff
@@ -281,6 +303,57 @@ private void handleRunTests(List<String> toolNames, List<String> sources) {
         }
     }
 
+
+    private void setUpGa4Ghv20Api() {
+        this.tooltesterConfig = new TooltesterConfig();
+        io.dockstore.openapi.client.ApiClient defaultApiClient = io.dockstore.openapi.client.Configuration.getDefaultApiClient();
+        defaultApiClient.setBasePath(this.tooltesterConfig.getServerUrl());
+        setGa4Ghv20Api(new Ga4Ghv20Api(defaultApiClient));
+    }
+
+    private void setUpExtendedGa4GhApi() {
+        this.tooltesterConfig = new TooltesterConfig();
+        io.dockstore.openapi.client.ApiClient defaultApiClient = io.dockstore.openapi.client.Configuration.getDefaultApiClient();
+        defaultApiClient.setBasePath(this.tooltesterConfig.getServerUrl());
+        defaultApiClient.setAccessToken(this.tooltesterConfig.getDockstoreAuthorizationToken());
+        setExtendedGa4GhApi(new ExtendedGa4GhApi(defaultApiClient));
+    }
+
+    private void runToolTesterOnWorkflows() throws InterruptedException {
+        setUpGa4Ghv20Api();
+        setUpExtendedGa4GhApi();
+
+        WorkflowList workflowsToRun = new WorkflowList(getGa4Ghv20Api(), getExtendedGa4GhApi());
+
+        for (WorkflowRunner workflow : workflowsToRun.getWorkflowsToRun()) {
+            workflow.runWorkflow();
+        }
+
+        List<WorkflowRunner> workflowsStillRunning = new ArrayList<>();
+        workflowsStillRunning.addAll(workflowsToRun.getWorkflowsToRun());
+
+        while (!workflowsStillRunning.isEmpty()) {
+            sleep(WAIT_TIME);
+            List<WorkflowRunner> workflowsToCheck = new ArrayList<>();
+            workflowsToCheck.addAll(workflowsStillRunning);
+            for (WorkflowRunner workflow: workflowsToCheck) {
+                if (workflow.isWorkflowFinished()) {
+                    workflowsStillRunning.remove(workflow);
+                }
+            }
+        }
+
+        printLine();
+        for (WorkflowRunner workflow: workflowsToRun.getWorkflowsToRun()) {
+            workflow.printRunStatistics();
+            printLine();
+        }
+
+        for (WorkflowRunner workflow: workflowsToRun.getWorkflowsToRun()) {
+            workflow.uploadRunInfo();
+        }
```

Comment:
Since you're uploading run info to Dockstore after all workflows have finished executing, you could group executions that have the same workflow and workflow version (like these [two executions](https://github.com/dockstore/dockstore-support/blob/7fef1d1ebae3b1b151423199d17ca8a03584577b/tooltester/src/main/java/io/dockstore/tooltester/runWorkflow/WorkflowList.java#L20-L21)) so you only end up call one POST for multiple executions of the same workflow version. Your `Executions` list would have multiple `Execution`s, one for each execution of the workflow version.

Reply:
I like this idea. Although I feel it would make the code a bit more complicated, and I don't know how common it would be to run a workflow multiple times.

I could create a ticket to add a parameter to the `WorkflowRunner` class that specifies how many times a workflow should be run, if that would be helpful.